### ISSUE TITLE
How to predict the class label of a new text in LSTM (which is not in training and testing set)?

### ISSUE THREAD

**Issue by AAMIBhavya** [fchollet/keras#2220]
How can I make use of the built model from [imdb_lstm.py](https://github.com/fchollet/keras/blob/master/examples/imdb_lstm.py) to predict the class label of a new input text which has no value corresponding to y_test ? Thanks in advance..

**Comment by qzshadow** [comment]
Using `model.predict(X)` method.

**Comment by AAMIBhavya** [comment]
@qzshadow Thanks..Is there no need of save and restore of the built model?

**Comment by qzshadow** [comment]
sure you have to save both the **model** and **weights** after training the networks, then when you use it, let's say you are giving samples X to get the predicts y, you need restore the **model** and load the **weights** into the model, then use this model to predict. but once the model is loaded into the memory, you could use different Xs to get different ys, no need to restore the model for each X.

**Comment by AAMIBhavya** [comment]
Thanks..But actually I am new to this field. If you get time, can you please help me regarding this **save** and **restore** of the model?

**Comment by qzshadow** [comment]
See http://keras.io/faq/#how-can-i-save-a-keras-model for details

**Comment by philipperemy** [comment]
Be sure that your vocabulary includes the new words as well. Otherwise you may have OOV (out of vocabulary) when you present your unseen data to your data processing layer before your LSTM crunches it.

**Comment by AAMIBhavya** [comment]
@qzshadow Thanks.. I will try

@philipperemy Thank you for the insight provided..
